packages/features/webhooks/lib/test/WebhookService.test.ts

```diff
@@ -0,0 +1,141 @@
+        appId: "appId",
+        payloadTemplate: "payloadTemplate",
+      },
+    ];
+    vi.mocked(getWebhooks).mockResolvedValue(mockWebhooks);
+
+    // Has to be called with await due to the iffi being async
+    const service = await new WebhookService(mockOptions);
+
+    expect(service).toBeInstanceOf(WebhookService);
+    expect(await service.getWebhooks()).toEqual(mockWebhooks);
+    expect(getWebhooks).toHaveBeenCalledWith(mockOptions);
+  });
+
+  // it("should send payload to all webhooks", async () => {
+  //   const mockWebhooks = [
+  //     {
+  //       id: "webhookId",
+  //       subscriberUrl: "url",
+  //       secret: "secret",
+  //       appId: "appId",
+  //       payloadTemplate: "payloadTemplate",
+  //     },
+  //     {
+  //       id: "webhookId2",
+  //       subscriberUrl: "url",
+  //       secret: "secret2",
+  //       appId: "appId2",
+  //       payloadTemplate: "payloadTemplate",
+  //     },
+  //   ];
+  //   vi.mocked(getWebhooks).mockResolvedValue(mockWebhooks);
+  //   const service = await new WebhookService(mockOptions);
+  //
+  //   const payload = {
+  //     secretKey: "secret",
+  //     triggerEvent: "triggerEvent",
+  //     createdAt: "now",
+  //     webhook: {
+  //       subscriberUrl: "url",
+  //       appId: "appId",
+  //       payloadTemplate: "payloadTemplate",
+  //     },
+  //     data: "test",
+  //   };
+  //
+  //   await service.sendPayload(payload as any);
+  //
+  //   expect(sendOrSchedulePayload).toHaveBeenCalledTimes(mockWebhooks.length);
+  //
+  //   mockWebhooks.forEach((webhook) => {
+  //     expect(sendOrSchedulePayload).toHaveBeenCalledWith(
+  //       webhook.secret,
+  //       mockOptions.triggerEvent,
+  //       expect.any(String),
+  //       webhook,
+  //       payload
+  //     );
+  //   });
+  // });
+  //
+  // it("should log error when sending payload fails", async () => {
+  //   const mockWebhooks = [
+  //     {
+  //       id: "webhookId",
+  //       subscriberUrl: "url",
+  //       secret: "secret",
+  //       appId: "appId",
+  //       payloadTemplate: "payloadTemplate",
+  //     },
+  //   ];
+  //   vi.mocked(getWebhooks).mockResolvedValue(mockWebhooks);
+  //
+  //   const logError = vi.fn();
+  //
+  //   (sendOrSchedulePayload as any).mockImplementation(() => {
+  //     throw new Error("Failure");
+  //   });
+  //
+  //   const service = new WebhookService(mockOptions);
+  //
+  //   const payload = {
+  //     secretKey: "secret", triggerEvent: "triggerEvent", createdAt: "now", webhook: {
+  //       subscriberUrl: "url",
+  //       appId: "appId",
+  //       payloadTemplate: "payloadTemplate"
+  //     },
+  //     data: "test"
+  //   };
+  //
+  //   await service.sendPayload(payload as any);
+  //
+  //   expect(logError).toHaveBeenCalledWith(
+  //     `Error executing webhook for event: ${mockOptions.triggerEvent}, URL: ${mockWebhooks[0].subscriberUrl}`,
+  //     JSON.stringify(new Error("Failure"))
+  //   );
+  // });
```

Comment:
Couldnt get this working as i wasnt sure how to spy on logger.getSublogger.error